src/test/java/com/epam/page/object/generator/integrationalTests/MainIntegrationTest.java

```diff
@@ -0,0 +1,283 @@
+            return false;
+        }
+    };
+
+    private boolean checkLocatorUniqueness;
+    private boolean forceGenerateFiles;
+    private String inputJavaFileTest;
+    private String inputJavaFileManual;
+    private String generatedClassFullName;
+    private String manualClassFullName;
+    private String jsonSourcePath;
+    private String url;
+    private Class testClass;
+    private Class manualClass;
+
+    private static List<File> filesList = new ArrayList<>();
+    private List<TestClassDTO> classesList;
+
+    @SuppressWarnings("Duplicates")
+    private PageObjectsGenerator initPog(String jsonPath, String url,
+                                         boolean checkLocatorUniqueness,
+                                         boolean forceGenerateFiles) throws IOException {
+        List<String> urls = new ArrayList<>();
+        urls.add(url);
+        SupportedTypesContainer bc = new SupportedTypesContainer();
+        JsonRuleMapper parser = new JsonRuleMapper(new File(jsonPath), new ObjectMapper());
+        XpathToCssTransformation xpathToCssTransformation = new XpathToCssTransformation();
+        JavaPoetAdapter javaPoetAdapter = new JavaPoetAdapter(bc, xpathToCssTransformation);
+
+        ValidatorsStarter validatorsStarter = new ValidatorsStarter(bc);
+        validatorsStarter.setCheckLocatorsUniqueness(checkLocatorUniqueness);
+
+        PageObjectsGenerator pog = new PageObjectsGenerator(parser, validatorsStarter,
+                javaPoetAdapter, RESOURCE_DIR, urls, PACKAGE_TEST_NAME);
+        pog.setForceGenerateFile(forceGenerateFiles);
+        return pog;
+    }
+
+    public MainIntegrationTest(List<TestClassDTO> list) throws MalformedURLException {
+        this.classesList = list;
+    }
+
+    @Parameters
+    public static Iterable<Object[]> data() throws IOException, ClassNotFoundException {
+
+        List<File> caseDirs = getMainDirsNames();
+        List<Object[]> testParameters = new ArrayList<>();
+
+        for (File caseDir : caseDirs) {
+            List<TestClassDTO> objectList = new ArrayList<>();
+            Properties caseProperties = new Properties();
+
+            List<File> insideDirs = (List<File>) FileUtils.listFilesAndDirs(
+                    caseDir,
+                    IO_FILE_FILTER,
+                    TrueFileFilter.INSTANCE
+            );
+            List<File> caseProps = (List<File>) FileUtils.listFiles(caseDir, null, false);
+            caseProperties.load(new FileInputStream(caseProps.get(0)));
+            insideDirs.remove(0);
+            for (File insideDir : insideDirs) {
+                filesList = (List<File>) FileUtils.listFiles(
+                        insideDir,
+                        TrueFileFilter.INSTANCE,
+                        TrueFileFilter.INSTANCE
+                );
+                for (File classFile : filesList) {
+                    String[] s = insideDir.getPath().split("[\\\\/]");
```

Comment:
Please, add commentaries to any user regular expressions. Commentaries should provide examples of input data and result of regular expression execution.